devito/mpi/halo_scheme.py

```diff
@@ -428,13 +428,24 @@ def compute_local_indices(f, dims, ispace, scope):
     loc_indices = {}
     for d in dims:
         try:
-            func = max if ispace.is_forward(d.root) else min
+            func = Max if ispace.is_forward(d.root) else Min
         except KeyError:
             raise HaloSchemeException("Don't know how to build a HaloScheme as `%s` "
                                       "doesn't appear in `%s`" % (d, ispace))
         candidates = [i[d] for i in scope.getreads(f) if not is_integer(i[d])]
         if d.is_Stepping:
-            loc_indices[d] = func(candidates, key=lambda i: i.origin - d)
+            loc_indices[d] = select(candidates, func=func, key=lambda i: i.origin - d)
         else:
-            loc_indices[d] = func(candidates, key=lambda i: i-d)
+            loc_indices[d] = select(candidates, func=func, key=lambda i: i - d)
     return loc_indices
+
+
+def select(candidates, func, key):
```

Comment:
This smells like an overkill. Do we really need a whole new function to select from `candidates`? That sounds odd

Reply:
You're totally right!! Refactored using a dictionary.